modules/common/src/main/java/unnamed/mmo/lang/LanguageProvider.java

```diff
@@ -0,0 +1,47 @@
+package unnamed.mmo.lang;
+
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.TextReplacementConfig;
+import org.jetbrains.annotations.NotNull;
+import unnamed.mmo.util.ComponentUtil;
+
+import java.io.IOException;
+import java.util.Properties;
+import java.util.regex.Pattern;
+
+/**
+ * Naive component translation system.
+ * <p>
+ * Should be replaced with adventure translation or something else in the future.
+ */
+public class LanguageProvider {
+    private static final Properties properties = new Properties();
+
+    static {
+        try {
+            properties.load(LanguageProvider.class.getResourceAsStream("/lang/en_US.properties"));
```

Comment:
You'll need an test language file, since this is producing an error when trying to run tests.

Reply:
Or perhaps better error handling when files are not found

Reply:
Just ignores a missing file, should be fine for now.